tests/gtests/api/test_memory.cpp

```diff
@@ -130,4 +130,117 @@ INSTANTIATE_TEST_SUITE_P(AllEngineKinds, memory_test_c_t, all_engine_kinds,
 INSTANTIATE_TEST_SUITE_P(AllEngineKinds, memory_test_cpp_t, all_engine_kinds,
         print_to_string_param_name_t());
 
+TEST(c_api_host_scalar_mem, TestHostScalarF32) {
```

Comment:
Random spot: scales/zero-points new API interface validation is missing.

Reply:
I've added simple testing similar to set_scales/etc.